### ISSUE TITLE
Geometry cutting of voronoi polygons for global IFD

### ISSUE THREAD

**Issue by schroedtert** [JuPedSim/jpscore#861]
Currently the room containing a particular measurement area will be detected and the resulting voronoi diagram will be cut at the borders of this room. Some issues arise from this when using the global IFD:
- room detection doesn't make sense when computing global IFD as the voronoi diagrams need to be cut at all walls not just one room
- as the walls might not have any thickness we can not differentiate between wall and door, which would mean that voronoi polygons will be cut at doors
- alternative: is the global IFD not the same as putting a measurement area in each room, which covers the whole area of that room?

**Comment by gjaeger** [comment]
> alternative: is the global IFD not the same as putting a measurement area in each room, which covers the whole area of that room?
@schroedtert No, the global IFD should be room independent. Voronoi area should depend only on the geometry, not in the definition of the rooms.

**Comment by chraibi** [comment]
I did not know that doors are relevant when cutting Voronoi diagrams.

>room detection doesn't make sense when computing global IFD as the voronoi diagrams need to be cut at all walls not just one room

Probably it makes sense with regards to commuting time?

**Comment by schroedtert** [comment]
> @schroedtert No, the global IFD should be room independent. Voronoi area should depend only on the geometry, not in the definition of the rooms.

This would just be a quick and dirty fix. But has the drawback, that artifacts will occur at the doors, as the voronoi diagrams will be cut there.



> I did not know that doors are relevant when cutting Voronoi diagrams.

They should not, but as we cut the voronoi polygon with the boundary of the geometry/room the doors are also included.



> Probably it makes sense with regards to commuting time?

From a first glance at the code, this would still lead to wrong results in a multiroom setup. Haven't verified it yet, though.
